pkgs/by-name/im/immich/package.nix

```diff
@@ -0,0 +1,241 @@
+{ lib
+, pkgs
+, buildNpmPackage
+, fetchFromGitHub
+, python3
+, nodejs
+, runCommand
+# build-time deps
+, pkg-config
+, makeWrapper
+, cmake
+, crane
+# runtime deps
+, ffmpeg
+, imagemagick
+, libraw
+, vips
+, perl
+}:
+let
+  buildNpmPackage' = buildNpmPackage.override { inherit nodejs; };
+  sources = lib.importJSON ./sources.json;
+  inherit (sources) version;
+
+  meta = with lib; {
+    description = "Self-hosted photo and video backup solution";
+    homepage = "https://immich.app/";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ jvanbruegge ];
+    inherit (nodejs.meta) platforms;
+  };
+
+  # The geodata website is not versioned,
+  # so we have to extract it from the container
+  geodata = runCommand "immich-geodata" {
+    outputHash = sources.geodata_hash;
+    outputHashMode = "recursive";
+  } ''
+    mkdir $out
+    echo "Downloading immich container image"
+    ${crane}/bin/crane export ghcr.io/immich-app/base-server-prod:${sources.container_tag} - \
+      | tar -xv -C "$out" --strip-components=3 usr/src/resources
+  '';
+
+  src = fetchFromGitHub {
+    owner = "immich-app";
+    repo = "immich";
+    rev = "v${version}";
+    inherit (sources) hash;
+  };
+
+  openapi = buildNpmPackage' {
+    pname = "immich-openapi-sdk";
+    inherit version;
+    src = "${src}/open-api/typescript-sdk";
+    inherit (sources.components."open-api/typescript-sdk") npmDepsHash;
+
+    installPhase = ''
+      runHook preInstall
+
+      npm config delete cache
+      npm prune --omit=dev --omit=optional
+
+      mkdir -p $out
+      mv package.json package-lock.json node_modules build $out/
+
+      runHook postInstall
+    '';
+  };
+
+  cli = buildNpmPackage' {
+    pname = "immich-cli";
+    inherit version;
+    src = "${src}/cli";
+    inherit (sources.components.cli) npmDepsHash;
+
+    nativeBuildInputs = [
+      makeWrapper
+    ];
+
+    preBuild = ''
+      rm node_modules/@immich/sdk
+      ln -s ${openapi} node_modules/@immich/sdk
+      # Rollup does not find the dependency otherwise
+      ln -s node_modules/@immich/sdk/node_modules/@oazapfts node_modules/
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out
+      mv package.json package-lock.json node_modules dist $out/
+
+      ls $out/dist
+
+      makeWrapper ${nodejs}/bin/node $out/bin/immich --add-flags $out/dist/index.js
+
+      runHook postInstall
+    '';
+
+    meta = meta // { mainProgram = "immich"; };
+  };
+
+  web = buildNpmPackage' {
+    pname = "immich-web";
+    inherit version;
+    src = "${src}/web";
+    inherit (sources.components.web) npmDepsHash;
+
+    inherit (cli) preBuild;
+
+    installPhase = ''
+      runHook preInstall
+
+      cp -r build $out
+
+      runHook postInstall
+    '';
+  };
+
+  python = python3;
```

Comment:
```suggestion
```
Normally we'd just use `python3` directly, we don't support python2 stuff anyway

Reply:
This was a left-over when there were extra python packages needed